src/task/task_environment.rs

```diff
@@ -127,15 +126,20 @@ impl<'p, D: TaskDisambiguation<'p>> SearchEnvironments<'p, D> {
                 .resolve(self.platform)
                 .find_map(|target| target.tasks.get(name))
             {
-                // None of the other environments can have this task. Otherwise, its still
-                // ambiguous.
+                // No other feature then default has the task defined.
                 if !self
                     .project
-                    .environments()
-                    .into_iter()
-                    .flat_map(|env| env.features(false).collect_vec())
-                    .flat_map(|feature| feature.targets.resolve(self.platform))
-                    .any(|target| target.tasks.contains_key(name))
+                    .manifest
+                    .parsed
+                    .features
+                    .iter()
+                    .filter(|(name, _feature)| name.as_str() != "default")
```

Comment:
Would be good to add a function to check if a feature is the default instead of comparing strings.

Reply:
Oops I think I got distracted